File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Cen/AlibabaCloud/SDK/ROS/CDK/Cen/IChildInstanceRouteEntryToAttachment.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Cen
{
    /// <summary>Represents a `ChildInstanceRouteEntryToAttachment`.</summary>
    [JsiiInterface(nativeType: typeof(IChildInstanceRouteEntryToAttachment), fullyQualifiedName: "@alicloud/ros-cdk-cen.IChildInstanceRouteEntryToAttachment")]
    public interface IChildInstanceRouteEntryToAttachment : AlibabaCloud.SDK.ROS.CDK.Core.IResource
    {
        /// <summary>Attribute CenId: The ID of the CEN instance.</summary>
        [JsiiProperty(name: "attrCenId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrCenId
        {
            get;
        }

        /// <summary>Attribute DestinationCidrBlock: The destination CIDR block of the route.</summary>
        [JsiiProperty(name: "attrDestinationCidrBlock", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrDestinationCidrBlock
        {
            get;
        }

        /// <summary>Attribute RouteTableId: The ID of the route table configured on the network instance.</summary>
        [JsiiProperty(name: "attrRouteTableId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrRouteTableId
        {
            get;
        }

        /// <summary>Attribute TransitRouterAttachmentId: The ID of the network instance connection.</summary>
        [JsiiProperty(name: "attrTransitRouterAttachmentId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrTransitRouterAttachmentId
        {
            get;
        }

        [JsiiProperty(name: "props", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-cen.ChildInstanceRouteEntryToAttachmentProps\"}")]
        AlibabaCloud.SDK.ROS.CDK.Cen.IChildInstanceRouteEntryToAttachmentProps Props
        {
            get;
        }

        /// <summary>Represents a `ChildInstanceRouteEntryToAttachment`.</summary>
        [JsiiTypeProxy(nativeType: typeof(IChildInstanceRouteEntryToAttachment), fullyQualifiedName: "@alicloud/ros-cdk-cen.IChildInstanceRouteEntryToAttachment")]
        internal sealed class _Proxy : DeputyBase, AlibabaCloud.SDK.ROS.CDK.Cen.IChildInstanceRouteEntryToAttachment
        {
            private _Proxy(ByRefValue reference): base(reference)
            {
            }

            /// <summary>Attribute CenId: The ID of the CEN instance.</summary>
            [JsiiProperty(name: "attrCenId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrCenId
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute DestinationCidrBlock: The destination CIDR block of the route.</summary>
            [JsiiProperty(name: "attrDestinationCidrBlock", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrDestinationCidrBlock
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute RouteTableId: The ID of the route table configured on the network instance.</summary>
            [JsiiProperty(name: "attrRouteTableId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrRouteTableId
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute TransitRouterAttachmentId: The ID of the network instance connection.</summary>
            [JsiiProperty(name: "attrTransitRouterAttachmentId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrTransitRouterAttachmentId
            {
                get => GetInstanceProperty<object>()!;
            }

            [JsiiProperty(name: "props", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-cen.ChildInstanceRouteEntryToAttachmentProps\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Cen.IChildInstanceRouteEntryToAttachmentProps Props
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Cen.IChildInstanceRouteEntryToAttachmentProps>()!;
            }

            /// <summary>The environment this resource belongs to.</summary>
            /// <remarks>
            /// For resources that are created and managed by the CDK
            /// (generally, those created by creating new class instances like Role, Bucket, etc.),
            /// this is always the same as the environment of the stack they belong to;
            /// however, for imported resources
            /// (those obtained from static methods like fromRoleArn, fromBucketName, etc.),
            /// that might be different than the stack they were imported into.
            /// </remarks>
            [JsiiProperty(name: "env", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.ResourceEnvironment\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.IResourceEnvironment Env
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.IResourceEnvironment>()!;
            }

            /// <summary>The stack in which this resource is defined.</summary>
            [JsiiProperty(name: "stack", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.Stack\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.Stack Stack
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.Stack>()!;
            }

            /// <summary>The construct tree node for this construct.</summary>
            [JsiiProperty(name: "node", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.ConstructNode\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode Node
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode>()!;
            }
        }
    }
}

File: multiple-languages/go/alicloudroscdkcen/IChildInstanceRouteEntryToAttachment.go
package alicloudroscdkcen

import (
	_jsii_ "github.com/aws/jsii-runtime-go/runtime"

	"github.com/alibabacloud-go/ros-cdk/alicloudroscdkcen/internal"
	"github.com/alibabacloud-go/ros-cdk/alicloudroscdkcore"
)

// Represents a `ChildInstanceRouteEntryToAttachment`.
type IChildInstanceRouteEntryToAttachment interface {
	alicloudroscdkcore.IResource
	// Attribute CenId: The ID of the CEN instance.
	AttrCenId() interface{}
	// Attribute DestinationCidrBlock: The destination CIDR block of the route.
	AttrDestinationCidrBlock() interface{}
	// Attribute RouteTableId: The ID of the route table configured on the network instance.
	AttrRouteTableId() interface{}
	// Attribute TransitRouterAttachmentId: The ID of the network instance connection.
	AttrTransitRouterAttachmentId() interface{}
	Props() *ChildInstanceRouteEntryToAttachmentProps
}

// The jsii proxy for IChildInstanceRouteEntryToAttachment
type jsiiProxy_IChildInstanceRouteEntryToAttachment struct {
	internal.Type__alicloudroscdkcoreIResource
}

func (j *jsiiProxy_IChildInstanceRouteEntryToAttachment) AttrCenId() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrCenId",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IChildInstanceRouteEntryToAttachment) AttrDestinationCidrBlock() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrDestinationCidrBlock",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IChildInstanceRouteEntryToAttachment) AttrRouteTableId() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrRouteTableId",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IChildInstanceRouteEntryToAttachment) AttrTransitRouterAttachmentId() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrTransitRouterAttachmentId",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IChildInstanceRouteEntryToAttachment) Props() *ChildInstanceRouteEntryToAttachmentProps {
	var returns *ChildInstanceRouteEntryToAttachmentProps
	_jsii_.Get(
		j,
		"props",
		&returns,
	)
	return returns
}

